packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts

```diff
@@ -268,6 +268,24 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
                 }
             }
         }));
+
+        toDispose.push(this.commands.registerCommand({
+            id: `${view.id}.focus`,
+            label: `Focus ${view.name} View`
+        }, {
+            execute: async () => {
+                const widget = await this.openView(view.id);
```

Comment:
I wonder why following does not work:
```
const widget = await this.openView(view.id);
if (widget) {
    this.shell.activateWidget(widget.id);
}
```

It will give the focus to the view, not to containing view container.

Reply:
Doing like just `this.shell.activateWidget(widget.id);` doesn't work here.

I had to find view container identifier (because it's registered in shell for the view) and use it to activate the view.

Reply:
I will need to check, i thought that each view is registered in the shell as well and the shell is capable to handle nested widgets. Focusing view container is not the same as focusing the view.

Reply:
Me too, but for that References View there is no reference view`s `widget.id` registered in `shell`'s tracker, 
In `shell` there is an array of tracked widgets that is used to search for a widget to activate by its id in `activateWidget(id: string)`..
So, the id of widget we need to activate is "plugin-view:references-view.tree" while that array contains "plugin-view-container:references-view" for Reference view - that's is why the References view cannot not activated by its view `widgett.id`.

Reply:
But anyway, `vscode-references-view` invokes `<view.id>.focus` command in order to activate the Reference View, so if we don't have such a command registered the plugin will not be able to activate the view.

Reply:
`activateWidget` creates a stack of all tracked widget, so it will be ['plugin-view:references-view.tree', 'plugin-view-container:references-view']. We need to activate all of them. If we activate only the last it will work only till view container has only one view (by luck). For stack it works in following order:
- activate the view container ('plugin-view-container:references-view')
- then ask the view container to activate the view ('plugin-view:references-view.tree')

If it does not work there is a bug in a view that it does not focus properly.

Reply:
I replaced it with proper activateWidget for recursive activation and could not observer the issue. When does not it work for you? Maybe there is a race between model is rendered and focus is requested?

Reply:
Focus command was working with no problem in Browser version but not working in Electron in this version:
```
            execute: async () => {
                const widget = await this.openView(view.id);
                if (widget) {
                    this.shell.activateWidget(widget.id);
                }
            }
```
I'm not sure when, but probably after some rebasing over the master branch, it stopped working, then I found that `plugin-view:references-view.tree` is not in the list of tracked widgets, it contains only `plugin-view-container:references-view` that's why i changed the handler to make activatoin by the view container identifier.

> I replaced it with proper activateWidget for recursive activation and could not observer the issue. When does not it work for you? Maybe there is a race between model is rendered and focus is requested?

This `Focus` command normally is invoked only if the References View is not open, so the handler should open the view and activate it. When I'm debugging it the View widget returned by `const widget = await this.openView(view.id);` is valid (not `null` or `undefined`), but its `id` is not on the list of tracked widgets, so the activation didn't worked.
If you tell that the widget `id` must be added to the list of tracked widgets, then probably we should look why it is not added there. (Without having it there even recursive activation probably will still be not full and correct?)

> I replaced it with proper activateWidget for recursive activation...

Did you made it locally in your bramch? Where can I look at it?